### PR TITLE
feat(bond): Phase 3.5 payout confirmation actions on mobile

### PR DESCRIPTION
**Pull request by Catrya**
- Add  and  to the Action enum
  - Add  helper (none/pending/acknowledged/completed) with retry support
  - Bond payout screen: hide form on acknowledged/completed, show info + single CLOSE button
  - My Trades badge: PAYOUT IN PROGRESS on acknowledged; order's real status on completed
  - Trade Details: extend the previous non-bond message with a bond-paid line; rate hint only if  is in history
  - : accept  default to
  - : keep tracked order untouched on bond acks; preserve current status (don't fall to default)
  - Notification extractor: return null for the two new actions (no SnackBar fires)
  - Exhaustive switches in notification mapper/item updated as no-ops (compiler requirement; functional wiring deferred)
  - l10n: new keys in en/es/it/de/fr
  - Tests: bond payout helpers 9→17 cases; regression test for

## Summary by CodeRabbit

* **New Features**
  * Added bond invoice acceptance and payout completion support with dedicated UI flows and localized messaging across five languages.
  * Introduced bond payout phase tracking for improved order state management.

* **Bug Fixes**
  * Made order status field optional to handle edge case payloads.

* **Tests**
  * Added comprehensive test coverage for bond payout phase transitions and order parsing.

[Review Change Stack](https://app.coderabbit.ai/change-stack/MostroP2P/mobile/pull/603?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lib/features/notifications/utils/notification_message_mapper.dart [hunk 1]
```diff
@@ -95,6 +95,8 @@ class NotificationMessageMapper {
       case mostro.Action.orders:
       case mostro.Action.lastTradeIndex:
       case mostro.Action.addBondInvoice:
+      case mostro.Action.bondInvoiceAccepted:
+      case mostro.Action.bondPayoutCompleted:
         return 'TODO: implement  title key if needed';
```
[the remark this answers is not in the document]
Comment (Catrya) [reply lib/features/notifications/utils/notification_message_mapper.dart:100]: Not blocking. getTitleKey is unreachable for these actions because notification_data_extractor.dart (this PR) returns null for both, which short-circuits both foreground SnackBar and background push paths before any title lokup happens. Same pattern as addBondInvoice (merged in PR #596). The TODO string is dead code; wiring real notification text is explicitly deferred to a follow-up PR.

File: lib/l10n/intl_fr.arb [hunk 2]
```diff
@@ -1588,5 +1588,10 @@
     "addBondInvoiceSubmitButton": "ENVOYER LA FACTURE",
     "addBondInvoiceFailedToSubmit": "Échec de l'envoi de la facture. Veuillez réessayer.",
     "bondPayoutBadge": "PAIEMENT EN ATTENTE",
+    "bondPayoutInProgressBadge": "PAIEMENT EN COURS",
+    "bondInvoiceAcceptedMessage": "Mostro a accepté la facture Lightning que vous avez fournie pour recevoir votre part du dépôt anti-abus. Veuillez patienter jusqu'à ce que le paiement soit terminé.",
+    "bondPayoutCompletedMessage": "Vous avez reçu le paiement de votre part du dépôt anti-abus.",
+    "bondPayoutCompletedWithRating": "Vous avez reçu le paiement de votre part du dépôt anti-abus. Vous pouvez évaluer votre contrepartie.",
+    "bondPayoutCloseButton": "FERMER",
```
[the remark this answers is not in the document]
Comment (Catrya) [reply lib/l10n/intl_fr.arb:1595]: Not blocking. The convention cited doesn't match repo practice: PR #596 (immediate Phase 3 precedent, same author, same feature family) added 11 bond keys with full fr/de translations. CLAUDE.md guidance is "Always use localized strings" and CI accepts populated values. If we want to switch to placeholder-only fr/de for new features, that's a policy change to discuss separately.

**Review by AndreaDiazCorreia — APPROVED** [review]
tACK
